addons/mail/static/src/js/owl/component/chat_window_manager.js

```diff
@@ -72,6 +72,16 @@ class ChatWindowManager extends owl.store.ConnectedComponent {
         return index < this.storeProps.computed.visible.length - 1;
     }
 
+    saveChatWindowsStates() {
+        const states = {};
+        for (const chatWindowLocalId in this.refs) {
+            states[chatWindowLocalId] = this.refs[chatWindowLocalId].getState();
```

Comment:
This will crash with the "new message" chat window, since it has no thread.
Also, if we add more refs in chat window manager template in the future, this will crash.
Maybe all chat window refs should be prefixed, e.g. `chatWindow_[chatWindowLocalId]`.

With changes above, this function would look like this:
```js
saveChatWindowsStates() {
   this.dispatch('updateChatWindowManager', {
      storedChatWindowStates: Object
         .entries(this.refs)
         .reduce((acc, [refId, ref]) => {
            if (!refId.startsWith('chatWindow_')) {
               return acc;
            }
            if (ref.props.chatWindowLocalId === 'new_message') {
               return acc;
            }
            return {
               ...acc,
               [ref.props.chatWindowLocalId]: ref.getState(),
            };
         }, {});
      },
   });
}
```

Reply:
So, I need to change the way chat windows refs are made ? I'm not sure to understand what I have to do ?

Reply:
Yes, something like `t-ref=" chatWindow_{{ data.chatWindowLocalId }}"` in the template.
Don't forget to adapt other uses of ref, like in `_handleAutofocus` of chat window manager

Reply:
Okay I'll check this, thanks :D